src/App.php

```diff
@@ -1063,25 +1072,50 @@ function () {
      */
     protected function outputResponseUnsafe(string $data, array $headersNew): void
     {
+        $isSse = $this->response->getHeaderLine('Content-Type') === 'text/event-stream';
+
+        if (headers_sent() && $isSse) {
+            echo $data;
+
+            return;
+        }
+
         $isCli = \PHP_SAPI === 'cli'; // for phpunit
 
         if (!headers_sent() || $isCli) {
             foreach ($headersNew as $k => $v) {
                 if (!$isCli) {
                     if ($k === self::HEADER_STATUS_CODE) {
-                        http_response_code($v === (string) (int) $v ? (int) $v : 500);
+                        $this->response = $this->response->withStatus($v === (string) (int) $v ? (int) $v : 500);
```

Comment:
if response is immutable, then will have too much overhead, all headers needs to be set at once, eg. response should be build and set once per `outputResponseUnsafe` call

Reply:
https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-7-http-message.md
The interface is clear add a single header and gets a new response with the new header, that's it, a different approach will do what I write there and I have a fear that creating a totally new response as no reason and is bad for future implementation (ex. JsSse) where i need to setup a response before creation.

Reply:
so for the 1st time (and it is normally called only once) constructor can be used - https://github.com/Nyholm/psr7/blob/master/src/Response.php#L42

https://github.com/Nyholm/psr7/blob/master/src/MessageTrait.php#L81 `withHeader` clones all the time

Reply:
Ok, it works for sure, but behind is nearly the same code : https://github.com/Nyholm/psr7/blob/fafc2faf8d49b979f922fe416dc8805462dc1fe8/src/MessageTrait.php#L138

Can be a problem because interface don't define a contract for `__construct`, so construct can be different from an implementation to another

Reply:
You are right, we must follow the interface (as the optimized constructor is not present).

On the otherside, if this is expected to be called only once per app lifecycle we should actually instance the response class here and then assign. `responseClass` is not helpfull at all, if middleware needs a specific class, it should convert from our (PSR-7 Response interfaced) response object.

Reply:
In my experience, whatever response we use if implements ResponseInterface is OK, so from what i see nyholm is the much stricter response library, i think we can stick to it and think is not our problem until someone will raise an issue, i think is a remote case because is the beauty of working with interface.
So we can remove the $responseClass and raise visibility to protected of $response to allow DI, what you think?